arangod/GeneralServer/RestHandler.cpp

```diff
@@ -66,7 +67,14 @@ RestHandler::RestHandler(ArangodServer& server, GeneralRequest* request,
               .with<structuredParams::UrlName>(_request->fullUrl())
               .with<structuredParams::UserName>(_request->user())
               .share()),
-      _canceled(false) {}
+      _canceled(false) {
+  if (server.hasFeature<GeneralServerFeature>() &&
+      server.isEnabled<GeneralServerFeature>()) {
+    _requestBodySizeTracker = metrics::GaugeCounterGuard<std::uint64_t>{
+        server.getFeature<GeneralServerFeature>()._requestBodySize,
+        _request->rawPayload().size()};
```

Comment:
if the request has Content-Type: application/json, the request may consume additional memory for its internal `_vpackBuilder` if it is an HTTP request. We should also count this memory usage.